src/components/deck-gl-wrapper/utils.ts

```diff
@@ -0,0 +1,206 @@
+import type { AttributeStorageInfo } from "@loaders.gl/i3s";
+import type { Tile3D } from "@loaders.gl/tiles";
+
+import { load } from "@loaders.gl/core";
+import { I3SAttributeLoader } from "@loaders.gl/i3s";
+
+export type COLOR = [number, number, number, number];
+export type ColorsByAttribute = {
+  /** Feature attribute name */
+  attributeName: string;
+  /** Minimum attribute value */
+  minValue: number;
+  /** Maximum attribute value */
+  maxValue: number;
+  /** Minimum color. 3DObject will be colorized with gradient from `minColor to `maxColor` */
+  minColor: COLOR;
+  /** Maximum color. 3DObject will be colorized with gradient from `minColor to `maxColor` */
+  maxColor: COLOR;
+  /** Colorization mode. `replace` - replace vertex colors with a new colors, `multiply` - multiply vertex colors with new colors */
+  mode: "multiply" | "replace";
+};
+export type ColorsByAttributeResult = {
+  tile: Tile3D;
+  colors?: Array<any>;
+  colorsByAttribute?: ColorsByAttribute;
+};
+
+/**
+ * Modify vertex colors array to visualize 3D objects in a attribute driven way
+ * @param tile - tile to be colorized
+ * @param colorsByAttribute - custom colors patameters
+ * @param options - loader options
+ * @returns original tile, new colors array and custom colors parameters
+ */
+// eslint-disable-next-line max-statements
+export async function customizeColors(
```

Comment:
This function is very I3S specific. I don't think it will be approved to be in the generic Tile3DLayer

Reply:
As discussed it will be a part of application and provided to the layer as a prop